### PR TITLE
[3.5] fix tests after update on UI

### PR DESCRIPTION
**Pull request by Seraphym-Tuhai**
## Pull Request description

image

Fixed UI broken tests  and locators
image

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [ ] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [ ] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [ ] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [ ] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [ ] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [ ] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

**Review by YevhenBondarenko — APPROVED** [review]

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/base/AbstractDriverBaseTest.java [hunk 1]
```diff
@@ -101,12 +100,13 @@ public void teardown() {
...
-    public void openLocalhost() {
-        driver.get(getBaseUiUrl());
+    public String getJwtTokenFromLocalStorage() {
+        js = (JavascriptExecutor) driver;
+        return (String) js.executeScript("return window.localStorage.getItem('jwt_token');");
     }
 
-    public void openHomePage() {
-        driver.get(getBaseUiUrl() + "/home");
+    public void openLocalhost() {
```
Comment (smatvienko-tb) [inline msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/base/AbstractDriverBaseTest.java:108]: base url is something variable. not exactly localhost. I suggest setting another name for the method `openLocalhost() `

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/pages/CustomerPageElements.java [hunk 2]
```diff
@@ -31,18 +31,18 @@ public CustomerPageElements(WebDriver driver) {
...
     protected static final String EDIT_MENU_DASHBOARD_FIELD = "//input[@formcontrolname='dashboard']";
-    private static final String EDIT_MENU_DASHBOARD = "//div[@class='cdk-overlay-pane']//span/span";
-    private static final String MANAGE_CUSTOMERS_USERS_BTN = ENTITY + "/../..//mat-icon[contains(text(),' account_circle')]/../..";
-    private static final String MANAGE_CUSTOMERS_ASSETS_BTN = ENTITY + "/../..//mat-icon[contains(text(),' domain')]/../..";
-    private static final String MANAGE_CUSTOMERS_DEVICES_BTN = ENTITY + "/../..//mat-icon[contains(text(),' devices_other')]/../..";
-    private static final String MANAGE_CUSTOMERS_DASHBOARDS_BTN = ENTITY + "/../..//mat-icon[contains(text(),' dashboard')]/../..";
-    private static final String MANAGE_CUSTOMERS_EDGE_BTN = ENTITY + "/../..//mat-icon[contains(text(),' router')]/../..";
+    private static final String EDIT_MENU_DASHBOARD = "//div[@class='cdk-overlay-pane']//span/span[contains(text(),'%s')]";
+    private static final String MANAGE_CUSTOMERS_USERS_BTN = ENTITY + "/ancestor::mat-row//mat-icon[contains(text(),' account_circle')]";
```
Comment (smatvienko-tb) [inline msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/pages/CustomerPageElements.java:35]: As we discussed today, the best approach is to have unique ids in the page source that likely never change

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/assetProfileSmoke/CreateAssetProfileTest.java [hunk 3]
```diff
@@ -167,7 +167,7 @@ public void documentation() {
         String urlPath = "docs/user-guide/asset-profiles/";
 
         sideBarMenuView.openAssetProfiles();
-        profilesPage.allEntity().get(0).click();
+        profilesPage.profileNames().get(0).click();
```
Comment (smatvienko-tb) [inline msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/assetProfileSmoke/CreateAssetProfileTest.java:170]: assertThat `profilesPage.profileNames()` isNotEmpty() with an alias like " profile names" to have an understandable output instead OutOfRange exception

**Review by smatvienko-tb — CHANGES_REQUESTED** [review]
Hi! In general, your PR looks good. Some minor suggestions to improve are below.

File: msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/assetProfileSmoke/CreateAssetProfileTest.java (hunk 3 above)
Comment (Seraphym-Tuhai) [reply msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/assetProfileSmoke/CreateAssetProfileTest.java:170]: Validation occurs in this custom waiter method:
image

Comment (Seraphym-Tuhai) [reply msa/black-box-tests/src/test/java/org/thingsboard/server/msa/ui/tests/assetProfileSmoke/CreateAssetProfileTest.java:170]: image

**Pull request merged**